library/src/main/java/com/klinker/android/send_message/Transaction.java

```diff
@@ -533,9 +533,15 @@ public static MessageInfo getBytes(Context context, boolean saveMessage, String[
         sendRequest.setDate(Calendar.getInstance().getTimeInMillis() / 1000L);
 
         try {
-            sendRequest.setFrom(new EncodedStringValue(Utils.getMyPhoneNumber(context)));
+            String phoneNumber = Utils.getMyPhoneNumberFromSubscription(context, settings.getSubscriptionId());
+            if (!TextUtils.isEmpty(phoneNumber)) {
+                sendRequest.setFrom(new EncodedStringValue(phoneNumber));
+            } else if (!TextUtils.isEmpty(fromAddress)) {
+                sendRequest.setFrom(new EncodedStringValue(fromAddress));
+            }
         } catch (Exception e) {
             Log.e(TAG, "error getting from address", e);
+            sendRequest.setFrom(new EncodedStringValue(fromAddress));
```

Comment:
If you switch the if/else blocks as noted in a previous comment, this line is no longer needed. It will set the `fromAddress` as the very first thing, if one exists. 

I would delete this line.